ERROR toolbar

### Astro Info

```block
Astro                    v5.7.13
Node                     v18.20.3
System                   macOS (arm64)
Package Manager          npm
Output                   static
Adapter                  none
Integrations             @astrojs/tailwind
                         @astrojs/react
```

### If this issue only occurs in one browser, which browser is a problem?

Chrome, Brave, Firefox

### Describe the Bug

[ERROR] [toolbar] Failed to initialize dev toolbar app Menu (astro:home):
TypeError: Cannot read properties of undefined (reading 'latestAstroVersion')
    at createCanvas (http://localhost:4321/node_modules/astro/dist/runtime/client/dev-toolbar/apps/astro.js?v=c0f2c8f5:60:60)
    at Object.init (http://localhost:4321/node_modules/astro/dist/runtime/client/dev-toolbar/apps/astro.js?v=c0f2c8f5:15:5)
    at AstroDevToolbar.initApp (http://localhost:4321/node_modules/astro/dist/runtime/client/dev-toolbar/toolbar.js?v=c0f2c8f5:342:23)
    at http://localhost:4321/node_modules/astro/dist/runtime/client/dev-toolbar/toolbar.js?v=c0f2c8f5:273:39
    at Array.map (<anonymous>)
    at window.requestIdleCallback.timeout (http://localhost:4321/node_modules/astro/dist/runtime/client/dev-toolbar/toolbar.js?v=c0f2c8f5:273:21)

### What's the expected result?

Main astro page when running `npm run dev`

### Link to Minimal Reproducible Example

.

### Participation

- [ ] I am willing to submit a pull request for this issue.

--- a/.changeset/sixty-nights-drive.md
+++ b/.changeset/sixty-nights-drive.md
@@ -1,0 +1,5 @@
+---
+'astro': patch
+---
+
+Fixes a case where the dev toolbar would crash if it could not retrieve some essential data

--- a/packages/astro/src/runtime/client/dev-toolbar/apps/astro.ts
+++ b/packages/astro/src/runtime/client/dev-toolbar/apps/astro.ts
@@ -92,8 +92,7 @@ export default {
 				},
 			];
 
-			const hasNewerVersion = (window as DevToolbarMetadata).__astro_dev_toolbar__
-				.latestAstroVersion;
+			const { latestAstroVersion, version, debugInfo } = (window as DevToolbarMetadata).__astro_dev_toolbar__ ?? {}
 
 			const windowComponent = createWindowElement(
 				`<style>
@@ -336,14 +335,10 @@ export default {
 			<header>
 				<section>
 				${astroLogo}
-				<astro-dev-toolbar-badge badge-style="gray" size="large">${
-					(window as DevToolbarMetadata).__astro_dev_toolbar__.version
-				}</astro-dev-toolbar-badge>
+				<astro-dev-toolbar-badge badge-style="gray" size="large">${version}</astro-dev-toolbar-badge>
 				${
-					hasNewerVersion
-						? `<astro-dev-toolbar-badge badge-style="green" size="large">${
-								(window as DevToolbarMetadata).__astro_dev_toolbar__.latestAstroVersion
-							} available!</astro-dev-toolbar-badge>
+					latestAstroVersion
+						? `<astro-dev-toolbar-badge badge-style="green" size="large">${latestAstroVersion} available!</astro-dev-toolbar-badge>
 						`
 						: ''
 				}
@@ -384,7 +379,7 @@ export default {
 
 			copyDebugButton?.addEventListener('click', () => {
 				navigator.clipboard.writeText(
-					'```\n' + (window as DevToolbarMetadata).__astro_dev_toolbar__.debugInfo + '\n```',
+					'```\n' + debugInfo + '\n```',
 				);
 				copyDebugButton.textContent = 'Copied to clipboard!';
 

